src/toolchain/web-guide.md

```diff
@@ -0,0 +1,135 @@
+<!--
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+
+# Intro to Web Builds
+
+Web builds are a fair bit more to get started with compared to native builds.
+This will be a complete guide on how to get things compiled.
+Setting up a web server is out-of-scope for this guide, and best found elsewhere.
+
+- Web support with `gdext` is experimental and should be understood as such before proceeding.
+
+---
+
+
+## Installing Pre-Requisites
+
+- Install a nightly build of `rustc`, the `wasm32-unknown-emscripten` target for `rustc`, and `rust-src`.
+Assuming that Rust was installed with rustup, this is quite simple.
+
+  ```sh
+  rustup toolchain install nightly
+  rustup component add rust-src --toolchain nightly
+  rustup target add wasm32-unknown-emscripten --toolchain nightly
+  ```
+
+- Install Emscripten
+  - The simplest way to achieve this is to install `emsdk` from the git repo. Recommended to use version 3.1.39 for now[^1]
+
+  ```sh
+  git clone https://github.com/emscripten-core/emsdk.git
+  cd emsdk
+  ./emsdk install 3.1.39
+  ./emsdk activate 3.1.39
+  source ./emsdk.sh     (or ./emsdk.bat on windows)
+  ```
+
+  - It would also be HIGHLY recommended to follow the instructions in the terminal to add `emcc`[^2] to your `PATH`.
+    If not, it is necessary to manually `source` the `emsdk.sh` file in every new terminal prior to compilation.
+    This is platform-specific.
+
+[^1]: Note: Due to a bug with `emscripten`, the maximum version of `emcc`[^2] that can one compile `Godot` with is `3.1.39`.  `gdext` itself should be able to support the latest version of `emcc`, however, it may be a safer bet to stick to version `3.1.39`.
+[^2]: `emcc` is the name of Emscripten's compiler
+
+---
+
+
+## Project Configuration
+
+- Enable the `experimental-wasm` feature on `gdext` in the `Cargo.toml` file.
+  It is also strongly recommended to enable the `lazy-function-tables` feature to avoid an extremely long compile time with release builds.
+  It is likely to due a bug, but is still being investigated.  Edit the line to something like the following:
```

Comment:
Maybe a bit better (`gdext` appears as `godot` so that first line might be confusing in principle)

```suggestion
- Enable the `experimental-wasm` feature on `gdext` in the `Cargo.toml` file.
  It is also strongly recommended to enable the `lazy-function-tables` feature to avoid an extremely long compile time with release builds, which is still being investigated (likely the result of a bug).
  Edit the `godot` dependency line in that file to something like the following:
```